nsqlookup/engine.go

```diff
@@ -40,62 +41,68 @@ type EngineInfo struct {
 
 // The Engine interface must be implemented by types that are intended to be
 // used to power nsqlookup servers.
+//
+// Each method of the engine accepts a context as last argument which may be
+// used to cancel or set a deadline on the operation.
+// This is useful for engines that work we storage services accessed over the
+// network.
+// The context may be nil.
 type Engine interface {
 	// Close should release all internal state maintained by the engine, it is
 	// called when the nsqlookup server using the engine is shutting down.
 	Close() error
 
 	// RegisterNode is called by nsqlookup servers when a new node is attempting
 	// to register.
-	RegisterNode(node NodeInfo) error
+	RegisterNode(node NodeInfo, ctx context.Context) error
```

Comment:
any reason we put it as the last param?

usually `ctx` goes first, go's context docs mention that they might introduce tooling that expects the `ctx` to be the first param.